src/batch.js

```diff
@@ -137,31 +139,25 @@ export default class Batch extends Middleware {
     const results = []
     for (let ii = 0; ii < batch.length; ++ii) {
       let r = response.data[ii]
-
-      // Currently use the presence of "status_code" to know that
-      // something has gone wrong.
-      if (r.status_code && r.status_code >= 300) {
-        const data = {
-          status_code: r.status_code,
-          reason_phrase: r.reason_phrase,
-          headers: r.headers || {},
-          body: r.body
+      const data = r.body
+      let success = false
+      if (r.status_code) {
+        if (r.status_code < 300) {
+          success = true
         }
-        // TODO: Should ensure this is treated the same way as
-        //  non batched errors.
-        batch[ii].reject({
-          response: data,
-          data
-        })
-        results.push(data)
       }
-      else {
+      if (success) {
         batch[ii].resolve({
           response: r,
-          data: r.body
+          data,
+        })
+      } else {
+        batch[ii].reject({
+          response: r,
+          data: data || {errors: {detail: r.reason_phrase}},
         })
-        results.push(r.body)
       }
+      results.push(data)
```

Comment:
@scaredcat Should we be consistent and pass the generated 'error data' from 157 into here too?

Reply:
```
        if (!data) data = {errors: {detail: r.reason_phrase}}
        batch[ii].reject({
          response: r,
          data,
        })
```

Reply:
probably better to do it that way, yes.